frontend/src/cljs/imas_seamap/events.cljs

```diff
@@ -79,6 +82,33 @@
   {:db db/default-db
    :dispatch [:db-initialised]})
 
+(defn get-serialized-hashstate
+  [{:keys [db]} _]
+  (let [serialized-hashstate-url (get-in db [:config :serialized-hashstate-url])
+        shortcode                (get db :shortcode)]
+    (cond-> {:db db}
+      shortcode
+      (assoc
+       :http-xhrio
+       [{:method :get
+         :uri             serialized-hashstate-url
+         :response-format (ajax/json-response-format {:keywords? true})
+         :on-success      [:got-serialized-hashstate]
+         :on-failure      [:ajax/default-err-handler]}])
+      (not shortcode)
+      (assoc :dispatch [:got-serialized-hashstate]))))
```

Comment:
I haven't tried to follow this exactly, but two impressions:

- (minor stylistic) this looks like another cond-masquerading-as-an-"if" to me
- I might be reading this wrong, but I'm quite leery of triggering the :got-... event when you didn't request it (perhaps it's a naming thing).  Without looking into how tricky, initial load has either no existing state / it has an existing saved state / it has a short-code that we need to retrieve.  They're all separate execution paths, even if they wind up in the same place.  (the :seen-any-of clause for async-flow looks useful)

Reply:
(in fact looking at the got-serialised-hashstate below, and really a lot of this code, it would be simplified if each case was a separate path; there's a lot of "cond->" followed by cases for whether there's a short code or not, etc, which just makes it hard to follow.

Reply:
I agree. I'll rewrite those then 👍